esmvalcore/_recipe.py

```diff
@@ -433,9 +433,25 @@ def _get_fx_files(variable, fx_info, config_user):
         fx_files = _get_input_files(fx_info, config_user)[0]
 
     # Flag a warning if no files are found
+    # CMIP6 special case:
+    # Before completely giving up, try looking for fx files in piControl:
+    # a lot of CMIP6 models have fx data only in piControl
     if not fx_files:
         logger.warning("Missing data for fx variable '%s'",
                        fx_info['short_name'])
+        if fx_info['project'] == 'CMIP6':
```

Comment:
I think you need to change the logic here:

```py
if not fx_files and fx_info['project'] == 'CMIP6':
    # do your thing
    # do NOT log warning if not successful
if not fx_files:
    # log the existing warning
```

With the current version, the warning about missing data is logged, even if the in the following iteration files are found for `piControl`, which is very confusing!

Reply:
yes exactly, imo the warning needs to be there since we switch experiments, if the user defines a preferred experiment they need to be aware that fx files were not found for that one, and instead the code tried to look for piControl ones

Reply:
Ok, how about:


```py
if not fx_files and fx_info['project'] == 'CMIP6':
    # do your thing
    # if files found, output warning that exp has changed
if not fx_files:
    # log the existing warning
```

As I said, right now it gives a warning that no files are found, then only says in debug that `piControl` is searched now, and if actually files are found, nothing else is returned. So if your log level is not `debug`, you just get the message about missing data, nothing else (even though the `piControl` data has been found)

Reply:
true, forgot those are debug! yeah, lemme change it :+1: